src/BuildScriptGeneratorCli/Commands/BuildpackDetectCommand.cs

```diff
@@ -0,0 +1,82 @@
+// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+using Microsoft.Oryx.Common;
+
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli
+{
+    [Command("buildpack-detect", Description = "Determines whether Oryx can be applied as a buildpack to " +
+        "an app in the current working directory.")]
+    internal class BuildpackDetectCommand : CommandBase
+    {
+        public const int DetectorFailCode = 100;
```

Comment:
curious if there will be any other failcode with 100

Reply:
what do you mean?

Reply:
makes sense now .. thanks for adding the comment